var/spack/repos/builtin/packages/camellia/package.py

```diff
@@ -0,0 +1,59 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Camellia(CMakePackage):
+    """Camellia: user-friendly MPI-parallel adaptive finite element package,
+       with support for DPG and other hybrid methods, built atop Trilinos.
+    """
+
+    homepage = "https://bitbucket.org/nateroberts/Camellia"
+    url      = "https://bitbucket.org/nateroberts/camellia.git"
+
+    maintainers = ['nvrober-sandia']
+
+    version('master',
+            git='https://bitbucket.org/nateroberts/camellia.git', branch='master')
+
+    depends_on('trilinos~shared+amesos+amesos2+anasazi+belos+epetra+epetraext+exodus+ifpack+ifpack2+intrepid+intrepid2+kokkos+ml+muelu+sacado+shards+teuchos+tpetra+zoltan+mumps+superlu-dist+hdf5+zlib+pnetcdf@master,12.12.1:')
```

Comment:
This seems... oddly specific.

Reply:
In creating this, I was following Nalu's package.py, which has a similar list; I modified it to include all the packages that Camellia depends on, as well as a couple on which we expect to depend in the future.  Is there a better way to do this?

Reply:
The problem is that Spack's `depends_on` isn't a suggestion, it is a requirement. Basically, this means that `camellia` can _only_ be installed with `trilinos` with _exaclty_ these variants enabled and disabled.

The better technique is to figure out which Boost libraries are actually required. If you just need the bare minimum of boost, `depends_on('boost')` is sufficient. If you need `epetra`, then `depends_on('boost+epetra)` is sufficient. If it can't be built with `shared` libraries for some reason, then `depends_on('boost~shared')` is sufficient. Any more specific than that and you're forcing someone to reinstall boost even though there installation would suffice.

Reply:
Ah yes, `boost` -> `trilinos`. `boost` is another package with a million variants, I guess that's why I got confused.